Sources/SwiftyRedis/PubSub.swift

```diff
@@ -157,21 +157,30 @@ public struct PubSubMessage: FromRedisValue {
             array = array.reversed()
             let message_type = try String(array.popLast())
 
-            channel = try String(array.popLast())
 
             switch message_type {
-            case "message", "pmessage", "smessage":
-                let payload = try String(array.popLast())
-
-                if let actual_payload = array.popLast() {
-                    channel = payload
-                    type = try .message(pattern: channel, payload: String(actual_payload))
```

Comment:
I was kind of trying to make this as covering as possible, so `"message", "pmessage", "smessage"` all in one case, but I could admit that this is somewhat hard to read/understand.